### PR TITLE
Reduce complexity of validateAuthenticateRoutePermission function

### LINKED ISSUE #73: Fix "method_complexity" issue in app/javascript/src/routes/index.js
Function `validateAuthenticateRoutePermission` has a Cognitive Complexity of 7 (exceeds 5 allowed). Consider refactoring.

https://codeclimate.com/github/chatwoot/chatwoot/app/javascript/src/routes/index.js#issue_5d87480387cf190001000189

### PR DESCRIPTION
**Pull request by fluffynuts**
fixes #73 

I don't see javascript tests in this project, however, I did run testing over the old and new functions to validate behavior. See attached.
[chatwoot-validate-auth-tests.zip](https://github.com/chatwoot/chatwoot/files/3679744/chatwoot-validate-auth-tests.zip)

### CODE REVIEW

**Comment by pranavrajs** [comment]
@fluffynuts Thanks for the PR. There are no javascript tests now. It’s good to initialise them as soon as possible. I will test this manually for now and merge.

**Comment by fluffynuts** [comment]
@pranavrajs if you raise another issue, I'll gladly get the scaffolding in place for testing (:

File: app/javascript/src/routes/index.js [hunk 1]
```diff
@@ -53,31 +53,47 @@ const authIgnoreRoutes = [
...
+    protected: true,
+    loggedIn: false,
+    handler: () => 'login',
+  },
+  {
+    protected: true,
+    loggedIn: true,
+    handler: to => {
+      const user = auth.getCurrentUser();
+      const isAccessible = routeIsAccessibleFor(to, user.role);
```
Comment (pranavrajs) [inline app/javascript/src/routes/index.js]: This should be `to.name`, right? @fluffynuts

Comment (pranavrajs) [reply app/javascript/src/routes/index.js]: Never mind you are already passing the name to the handler.

**Comment by pranavrajs** [comment]
@fluffynuts Could you please share your details here? We would like to send you a sticker pack for creating the PR to our repo. We will start shipping it by the end of October.

**Comment by fluffynuts** [comment]
Thanks for merging (: I'm still looking into the issue to add a unit test framework to this project. I have unit tests running, but without proper babel support (eg `import 'some-lib'`) fails, so I must be doing Something Wrong with Jest in this regard. Will open the PR when I figure it out (:

**Pull request merged**